components/Typography/docs/dynamic-type.md

```diff
@@ -0,0 +1,33 @@
+### Dynamic Type
+
+#### Overview
+Typography library provides APIs that allow users to support Dynamic Type feature in Material Components. For those components with Dynamic Type feature supported, text size changes based on the Dynamic Type setting provided by the user.
+
+To support Dynamic Type in your component, you need to set up text styles represented by `MDCFontTextStyle` on text, such as `MDCFontTextStyleBody1` or `MDCFontTextStyleTitle`. These text styles describe how text size should be adjusted when Dynamic Type setting changes.
```

Comment:
In this introduction, focus on describing at a high level the two essential things a client needs to do in order to enable Dynamic Type behaviors on a component:

1. Enable mdc_adjustsFontForContentSizeCategory on the component instance.
2. Set a scaled font using the MDCFontScaler APIs.

Reply:
The first one is documented in https://github.com/material-components/material-components-ios/pull/7431/files#diff-b8352f12ab0e41f2d34e03bd34d87269R33. I feel directly putting them together confuse the user. And they are not parallel, which is why I separate #1 into a separate section. #2 is what this paragraph describes. How do you feel? We can talk in person.

Reply:
And I think this brief paragraph contains all information about MDCFontScaler APIs, so we don't need to provide a higher level of the information.

Reply:
We may need to chat in person, but I'll make one more effort here in case we don't need to:

My understanding is that a client needs to do both of the two things I outlined above. This paragraph only discusses the second, with the first being discussed near the end of the document. I think it's important for the reader to know that they need to both enable the first property and provide a font with associated scaling curves.

Reply:
My understanding is the same. I lean on restructuring it to make these two parts closer after reading your comment there. We can review the structure from there after I push the changes.

Thanks.

Reply:
Addressed in 32e2339180fe1be49a20bb7e3931d105f363a0dd